server.go

```diff
@@ -86,29 +99,42 @@ func (s *StubServer) HandleRequest(w http.ResponseWriter, r *http.Request) {
 	start := time.Now()
 	fmt.Printf("Request: %v %v\n", r.Method, r.URL.Path)
 
+	// Prepare internal error as it's the most commonly used here
+	stripeError := &ResponseError{
+		ErrorInfo: ErrorInfo{
+			Message: internalServerError,
+			Type:    "invalid_request_error",
+		},
+	}
+
 	auth := r.Header.Get("Authorization")
 	if !validateAuth(auth) {
-		writeResponse(w, r, start, http.StatusUnauthorized,
-			fmt.Sprintf(invalidAuthorization, auth))
+		stripeError.ErrorInfo.Message = fmt.Sprintf(invalidAuthorization, auth)
+		writeResponse(w, r, start, http.StatusUnauthorized, stripeError)
 		return
 	}
 
+	// Every response needs a Request-Id header except the invalid authorization
+	w.Header().Set("Request-Id", "req_123")
+
 	route := s.routeRequest(r)
 	if route == nil {
-		writeResponse(w, r, start, http.StatusNotFound, nil)
+		stripeError.ErrorInfo.Message = fmt.Sprintf(invalidRoute,
+			r.Method, r.URL.Path)
+		writeResponse(w, r, start, http.StatusNotFound, stripeError)
 		return
 	}
 
 	response, ok := route.operation.Responses["200"]
 	if !ok {
 		fmt.Printf("Couldn't find 200 response in spec\n")
```

Comment:
Would it make sense to have one function that both prints the error message and generates the error response?

Reply:
Made it better in https://github.com/stripe/stripe-mock/pull/32/commits/f9ad1e76482ce44b1eb9367cc8485dac93dd39a1 though the message is still generate first and then passed as a parameter. LMK if that fits the ask

Reply:
The issue is that the strings can have 0/1/N dynamic parameters (a bit like printf) so we'd have to rebuild a printf-like version right?

Something like `createStripeError(errorType string, errorMessagePattern string, params ...string)`.

Do you think it's worth it? Happy to implement if so!

Reply:
Another option would be something like `createInternalServerError(fmt.Sprintf("missing widget: %s", widget))`.

I don't have a strong opinion either way; I just wanted to raise the possibility. I'll leave it up to your judgement.

Reply:
Ah yes I did that at first but with the rule of rows no more than 80 characters then it ended up spanning 2 rows anyway.

For example:
```
	if route == nil {
		message := fmt.Sprintf(invalidRoute, r.Method, r.URL.Path)
		stripeError := createStripeError(typeInvalidRequestError, message)
		writeResponse(w, r, start, http.StatusNotFound, stripeError)
		return
	}
```
would be
```
	if route == nil {
		message := 
		stripeError := createStripeError(typeInvalidRequestError,
			fmt.Sprintf(invalidRoute, r.Method, r.URL.Path))
		writeResponse(w, r, start, http.StatusNotFound, stripeError)
		return
	}
```

Do you prefer the latter?

Reply:
Either is fine; I'll leave it to your judgement.

Reply:
Cool, I'll leave as is then because that's my personal preference but we can always improve this in a future PR if needed.